capsules/src/adc.rs

```diff
@@ -90,6 +172,39 @@ impl<'a, A: AdcSingle + 'a> Driver for ADC<'a, A> {
             2 => {
                 self.sample(data as u8, appid)
             },
+            3 => {
+                // Due to the 32-bit limit of the data parameter to the
+                // `command()' system call, only the lower 24 bits of
+                // interval are used, leaving 8 bits for CHANNEL.
+                let channel = (data & 0xFF) as u8;
+                let interval = (data >> 8) as u32;
+                if self.compute_interval(0) <= interval {
+                    self.sample_continuous(channel, interval, appid)
+                } else if self.compute_interval_fast(0) <= interval {
+                    self.sample_continuous_fast(channel, interval, appid)
+                } else {
+                    self.sample_continuous_very_fast(channel, interval, appid)
+                }
+            
+            },
+            4 => {
+                self.cancel_sampling()
+            },
+            5 => {
+                let interval = data as u32;
+                let precise_interval;
+                if self.compute_interval(0) <= interval {
+                    precise_interval = self.compute_interval(interval) 
+                } else if self.compute_interval_fast(0) <= interval {
+                    precise_interval = self.compute_interval_fast(interval) 
+                } else {
+                    precise_interval = self.compute_interval_very_fast(interval) 
+                }
+                self.app.each(|app| { 
+                    app.int_callback.map(|mut cb| cb.schedule(0, 0, precise_interval as usize));
+                });
```

Comment:
Instead of sending the interval value in a callback, you can use return `ReturnCode::SuccessWithValue{precise_interval}` instead.

If you were going to perform a callback, using `.each()` is probably the wrong way to go, as it would perform the callback for each application, most of which don't care about this interval value. Instead, you should use `.enter(appid, ...)` which will let you perform the callback on only the application that called the command.